keras_nlp/models/gpt_neox/gpt_neox_attention.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.models.gpt_neox.rotary_embedding import RotaryEmbedding
+from keras_nlp.utils.keras_utils import clone_initializer
+
+
+class GPTNeoXAttention(keras.layers.Layer):
+    def __init__(
+        self,
+        num_heads,
+        hidden_dim,
+        dropout=0.1,
+        max_sequence_length=512,
+        kernel_initializer="glorot_uniform",
+        bias_initializer="zeros",
+        rotary_pct=0.25,
```

Comment:
I guess one option is to document this the same as `max_wavelength` in our SinePositionEncoding layer. https://keras.io/api/keras_nlp/modeling_layers/sine_position_encoding/

I'm not sure it's the best name, but at least it will be consistent across the library. We could name these arguments `rotary_percentage` and `rotary_max_wavelength` here, and just `percentage` and `max_wavelength` on the rotary layer itself.

Reply:
fixed this !